### PR TITLE
Fix/bmp 1 fix internazionalization

### PR DESCRIPTION
**Pull request by Luca-Ferrera**
This PR aims to fix the [bug](https://makeitapp.atlassian.net/browse/BMP-1) with i18n. It also adds italian and english strings in order to improve the i18n. 

#### Checklist

- [x] your branch will not cause merge conflict with `master`
- [x] run `npm test`
- [x] tests are included
- [x] the documentation is updated or integrated accordingly with your changes
- [x] the code will follow the lint rules and style of the project
- [x] you are not committing extraneous files or sensible data

### CODE REVIEW

File: packages/api-explorer/__tests__/SchemaTabs.test.jsx [hunk 1]
```diff
@@ -97,7 +99,7 @@ describe('SchemaTabs', () => {
       element.update()
       const formattedMessage = element.find(FormattedMessage)
       expect(formattedMessage).toHaveLength(1)
-      expect(formattedMessage.prop('id')).toEqual('schemaTabs.missing.request')
+      expect(formattedMessage.prop('id')).toEqual('schemaTabs.missing.Request')
```
Comment (davidebianchi) [inline packages/api-explorer/__tests__/SchemaTabs.test.jsx:102]: Why the upper casing?

File: packages/api-explorer/__tests__/SchemaTabs.test.jsx [hunk 2]
```diff
@@ -131,7 +133,7 @@ describe('SchemaTabs', () => {
       element.update()
       const formattedMessage = element.find(FormattedMessage)
       expect(formattedMessage).toHaveLength(1)
-      expect(formattedMessage.prop('id')).toEqual('schemaTabs.missing.response')
+      expect(formattedMessage.prop('id')).toEqual('schemaTabs.missing.Response')
```
Comment (davidebianchi) [inline packages/api-explorer/__tests__/SchemaTabs.test.jsx:136]: Why the upper casing?

File: packages/api-explorer/src/Doc.jsx [hunk 3]
```diff
@@ -208,7 +208,7 @@ class Doc extends React.Component {
 
     return list && list.length !== 0 && showTitle ? (
       <ContentWithTitle
-        title='Select Content Type'
+        title={<FormattedMessage id={`doc.Select Content Type`} defaultMessage={'Select Content Type'} />}
```
Comment (davidebianchi) [inline packages/api-explorer/src/Doc.jsx:211]: The Id prop is not a correct id

File: packages/api-explorer/src/components/SchemaTabs/index.jsx [hunk 4]
```diff
@@ -56,7 +56,7 @@ export default class SchemaTabs extends Component {
         operation.responses ? (
           <ResponseSchema operation={operation} oas={oas} />
         )
-        : renderMissingSchema('response')
+        : renderMissingSchema('Response')
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/SchemaTabs/index.jsx:59]: Why response in upper case?

File: packages/api-explorer/src/components/SchemaTabs/index.jsx [hunk 5]
```diff
@@ -67,15 +67,17 @@ export default class SchemaTabs extends Component {
         operation.requestBody ? (
           <RequestSchema operation={operation} oas={oas} />
         )
-        : renderMissingSchema('request')
+        : renderMissingSchema('Request')
```
Comment (davidebianchi) [inline packages/api-explorer/src/components/SchemaTabs/index.jsx:70]: Also here

File: packages/api-explorer/src/index.jsx [hunk 6]
```diff
@@ -184,7 +184,7 @@ class ApiExplorer extends React.Component {
 
     const defaultOpenDoc = this.props.defaultOpenDoc ? this.props.defaultOpenDoc : '0'
     const defaultOpen = this.props.defaultOpen ? [defaultOpenDoc] : null
-    const localizedMessages = messages[this.props.i18n.locale] || messages[this.props.i18n.defaultLocale]
+    const localizedMessages = messages[this.props.i18n.locale] || messages[this.props.i18n.defaultLocale] || messages.en
```
Comment (davidebianchi) [inline packages/api-explorer/src/index.jsx:187]: Default locale should already be en. Why is it added the last or statement?

File: packages/api-explorer/src/components/SchemaTabs/index.jsx (hunk 5 above)
Comment (Luca-Ferrera) [reply packages/api-explorer/src/components/SchemaTabs/index.jsx:70]: Fixed everywhere

File: packages/api-explorer/src/index.jsx (hunk 6 above)
Comment (Luca-Ferrera) [reply packages/api-explorer/src/index.jsx:187]: It wasn't, it happens that passing `{ locale: 'en}` as prop overrides the default prop
```
i18n: {
    locale: 'en',
    defaultLocale: 'en',
  }
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,10 @@ All notable changes to this project will be documented in this file. Dates are d
 
 Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).
 
+### Unreleased
```
Comment (fredmaggiowski) [inline CHANGELOG.md:7]: Changelog is automatic here, no need to write update it manually :)

File: packages/api-explorer/src/JsonForm/index.jsx [hunk 8]
```diff
@@ -21,6 +22,9 @@ function configureJSONEditor(intl, setFormSubmissionListener) {
     JSONEditor.defaults.editors[key] = getCustomEditor(key);
   });
 
+  JSONEditor.defaults.languages.it = it
```
Comment (fredmaggiowski) [inline packages/api-explorer/src/JsonForm/index.jsx:25]: Don't we need to set the same for `en` too?

File: packages/api-explorer/src/index.jsx (hunk 6 above)
Comment (davidebianchi) [reply packages/api-explorer/src/index.jsx:187]: It override the `defaultLocale` prop? Here the code try to access using `locale`, `defaultLocale` if locale is not defined. But when one of those are not in i18n object?

File: packages/api-explorer/src/JsonForm/index.jsx (hunk 8 above)
Comment (Luca-Ferrera) [reply packages/api-explorer/src/JsonForm/index.jsx:25]: by default it is `en`

Comment (fredmaggiowski) [reply packages/api-explorer/src/JsonForm/index.jsx:25]: ok

File: packages/api-explorer/src/index.jsx (hunk 6 above)
Comment (Luca-Ferrera) [reply packages/api-explorer/src/index.jsx:187]: I was looking that if the `API-Portal website` use this component and pass as prop `i18n = { locale: 'en'}`  then the `defaultLocale` here is `undefined`. That's why I was saying that the `defaultLocale` prop is overridden

Comment (davidebianchi) [reply packages/api-explorer/src/index.jsx:187]: So the issue in `API-Portal website` is a not correct use of this library, I suppose. Is it possible to add the default locale in that website?

A fix for this library could be set the locale and defaultLocale as required, what do you think about it?

Comment (Luca-Ferrera) [reply packages/api-explorer/src/index.jsx:187]: yes, for sure I can add the default locale in the `API-Portal website`, also setting locale and defaultLocale as required should solve the problem!

**Pull request merged**